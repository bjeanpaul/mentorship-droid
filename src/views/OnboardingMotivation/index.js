import React, { PropTypes } from 'react';
import { View } from 'react-native';
import { BaseView, Header, Text, Pagination, ProgressBar,
  MultiLineTextInput } from 'src/components';
import styles from './styles';


const OnboardingMotivation = ({
  jobMotivation = '',
  onChangeText,
  onBackPress,
  onNextPress,
}) => {
  const isPaginationDisabled = jobMotivation.length === 0;
  return (
    <BaseView>
      <ProgressBar completed={4 / 7} />
      <Header>
        <Text style={Text.types.title}>Why did you become a mentor?</Text>
      </Header>

      <MultiLineTextInput
        value={jobMotivation}
        placeholder="Type your answer here"
        maxLength={50}
        onChangeText={(text) => onChangeText({ jobMotivation: text })}
      />

      <View style={styles.footer}>
        <Pagination
          onBackPress={onBackPress}
          onNextPress={onNextPress}
          disabled={isPaginationDisabled}
        />
      </View>
    </BaseView>
  );
};


OnboardingMotivation.propTypes = {
  jobMotivation: PropTypes.string,
  onChangeText: PropTypes.func.isRequired,
  onBackPress: PropTypes.func.isRequired,
  onNextPress: PropTypes.func.isRequired,
};


export default OnboardingMotivation;
