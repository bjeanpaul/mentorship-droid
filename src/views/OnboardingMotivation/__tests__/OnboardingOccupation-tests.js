import React from 'react';
import { noop } from 'lodash';
import OnboardingMotivation from 'src/views/OnboardingMotivation';


describe('OnboardingOccupation', () => {
  function createComponent(props = {}) {
    return (
      <OnboardingMotivation
        onNextPress={noop}
        onBackPress={noop}
        onChangeText={noop}
        {...props}
      />
    );
  }

  it('should render', () => {
    expect(render(createComponent())).toMatchSnapshot();
  });

  it('should enable pagination next button', () => {
    expect(render(createComponent({
      jobMotivation: 'Because I love children.',
    }))).toMatchSnapshot();
  });
});
