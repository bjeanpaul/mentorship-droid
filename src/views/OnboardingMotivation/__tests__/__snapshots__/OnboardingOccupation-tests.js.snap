exports[`OnboardingOccupation should enable pagination next button 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
        "flex": 1
      },
      undefined
    ]
  }>
  <View
    style={
      Object {
        "backgroundColor": "#e5e5e5",
        "flexDirection": "row",
        "height": 6
      }
    }>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#f77040"
          },
          Object {
            "flex": 0.6666666666666667
          }
        ]
      } />
    <View
      style={
        Object {
          "flex": 0.33333333333333326
        }
      } />
  </View>
  <View
    style={
      Array [
        Object {
          "height": 72,
          "justifyContent": "center",
          "paddingLeft": 16,
          "paddingRight": 16
        },
        Object {}
      ]
    }>
    <Text
      style={
        Array [
          Object {
            "color": "#003035",
            "fontFamily": "Brandon Text",
            "fontSize": 13,
            "textAlign": "center"
          },
          Object {
            "fontFamily": "Brandon Text_medium",
            "fontSize": 20,
            "fontWeight": "400"
          }
        ]
      }>
      Why did you become a mentor?
    </Text>
  </View>
  <View>
    <View
      style={
        Array [
          Object {
            "borderBottomWidth": 1,
            "borderColor": "#dfe5e6",
            "marginBottom": 27,
            "marginLeft": 24,
            "marginRight": 24
          },
          false
        ]
      }>
      <TextInput
        maxLength={50}
        multiline={true}
        onBlur={[Function onBlur]}
        onChange={[Function onChange]}
        onChangeText={[Function onChangeText]}
        onFocus={[Function onFocus]}
        placeholder="Type your answer here"
        placeholderTextColor="#c0cbcc"
        style={
          Array [
            Object {
              "color": "#003035",
              "fontFamily": "Brandon Text",
              "fontSize": 18,
              "lineHeight": 28,
              "marginBottom": 8,
              "padding": 0
            },
            Object {
              "height": 32
            }
          ]
        }
        underlineColorAndroid="transparent"
        value="" />
    </View>
    <Text
      style={
        Array [
          Object {
            "color": "#003035",
            "fontFamily": "Brandon Text",
            "fontSize": 13,
            "textAlign": "center"
          },
          Object {
            "color": "#c0cbcc",
            "fontSize": 14,
            "margin": 28,
            "marginBottom": 0,
            "marginTop": -20,
            "textAlign": "right"
          }
        ]
      }>
      0
      /
      50
    </Text>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "justifyContent": "flex-end"
      }
    }>
    <View
      style={
        Object {
          "flexDirection": "row",
          "margin": 16
        }
      }>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#ffbcbc",
              "borderColor": "red",
              "borderWidth": 1,
              "height": 100,
              "justifyContent": "center",
              "margin": 10,
              "width": 300
            },
            undefined
          ]
        }>
        <Text
          style={
            Object {
              "color": "#333333",
              "margin": 20
            }
          }>
          TouchableNativeFeedback is not supported on this platform!
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "borderRadius": 5,
              "justifyContent": "center"
            },
            Object {
              "backgroundColor": "transparent",
              "borderColor": "#f77040",
              "borderWidth": 1
            },
            Array [
              Object {
                "marginLeft": 0,
                "marginRight": 0
              },
              Object {
                "flex": 1
              }
            ],
            Object {
              "borderColor": "#dfe5e6"
            }
          ]
        }>
        <Text
          style={
            Array [
              Object {
                "color": "#003035",
                "fontFamily": "Brandon Text",
                "fontSize": 13,
                "textAlign": "center"
              },
              Array [
                Object {
                  "fontFamily": "Brandon Text_bold",
                  "fontSize": 13,
                  "fontWeight": "bold",
                  "letterSpacing": 1,
                  "marginBottom": 17,
                  "marginTop": 15
                },
                Object {
                  "color": "#f77040"
                },
                Object {
                  "color": "#dfe5e6"
                }
              ]
            ]
          }>
          NEXT
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`OnboardingOccupation should render 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
        "flex": 1
      },
      undefined
    ]
  }>
  <View
    style={
      Object {
        "backgroundColor": "#e5e5e5",
        "flexDirection": "row",
        "height": 6
      }
    }>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#f77040"
          },
          Object {
            "flex": 0.6666666666666667
          }
        ]
      } />
    <View
      style={
        Object {
          "flex": 0.33333333333333326
        }
      } />
  </View>
  <View
    style={
      Array [
        Object {
          "height": 72,
          "justifyContent": "center",
          "paddingLeft": 16,
          "paddingRight": 16
        },
        Object {}
      ]
    }>
    <Text
      style={
        Array [
          Object {
            "color": "#003035",
            "fontFamily": "Brandon Text",
            "fontSize": 13,
            "textAlign": "center"
          },
          Object {
            "fontFamily": "Brandon Text_medium",
            "fontSize": 20,
            "fontWeight": "400"
          }
        ]
      }>
      Why did you become a mentor?
    </Text>
  </View>
  <View>
    <View
      style={
        Array [
          Object {
            "borderBottomWidth": 1,
            "borderColor": "#dfe5e6",
            "marginBottom": 27,
            "marginLeft": 24,
            "marginRight": 24
          },
          false
        ]
      }>
      <TextInput
        maxLength={50}
        multiline={true}
        onBlur={[Function onBlur]}
        onChange={[Function onChange]}
        onChangeText={[Function onChangeText]}
        onFocus={[Function onFocus]}
        placeholder="Type your answer here"
        placeholderTextColor="#c0cbcc"
        style={
          Array [
            Object {
              "color": "#003035",
              "fontFamily": "Brandon Text",
              "fontSize": 18,
              "lineHeight": 28,
              "marginBottom": 8,
              "padding": 0
            },
            Object {
              "height": 32
            }
          ]
        }
        underlineColorAndroid="transparent"
        value="" />
    </View>
    <Text
      style={
        Array [
          Object {
            "color": "#003035",
            "fontFamily": "Brandon Text",
            "fontSize": 13,
            "textAlign": "center"
          },
          Object {
            "color": "#c0cbcc",
            "fontSize": 14,
            "margin": 28,
            "marginBottom": 0,
            "marginTop": -20,
            "textAlign": "right"
          }
        ]
      }>
      0
      /
      50
    </Text>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "justifyContent": "flex-end"
      }
    }>
    <View
      style={
        Object {
          "flexDirection": "row",
          "margin": 16
        }
      }>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#ffbcbc",
              "borderColor": "red",
              "borderWidth": 1,
              "height": 100,
              "justifyContent": "center",
              "margin": 10,
              "width": 300
            },
            undefined
          ]
        }>
        <Text
          style={
            Object {
              "color": "#333333",
              "margin": 20
            }
          }>
          TouchableNativeFeedback is not supported on this platform!
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "borderRadius": 5,
              "justifyContent": "center"
            },
            Object {
              "backgroundColor": "transparent",
              "borderColor": "#f77040",
              "borderWidth": 1
            },
            Array [
              Object {
                "marginLeft": 0,
                "marginRight": 0
              },
              Object {
                "flex": 1
              }
            ],
            Object {
              "borderColor": "#dfe5e6"
            }
          ]
        }>
        <Text
          style={
            Array [
              Object {
                "color": "#003035",
                "fontFamily": "Brandon Text",
                "fontSize": 13,
                "textAlign": "center"
              },
              Array [
                Object {
                  "fontFamily": "Brandon Text_bold",
                  "fontSize": 13,
                  "fontWeight": "bold",
                  "letterSpacing": 1,
                  "marginBottom": 17,
                  "marginTop": 15
                },
                Object {
                  "color": "#f77040"
                },
                Object {
                  "color": "#dfe5e6"
                }
              ]
            ]
          }>
          NEXT
        </Text>
      </View>
    </View>
  </View>
</View>
`;
